src/extensions/scratch3_boost/index.js

```diff
@@ -0,0 +1,1963 @@
+        degrees = Math.abs(MathUtil.clamp(degrees, -360000, 360000));
+
+        const motors = [];
+        this._forEachMotor(args.MOTOR_ID, motorIndex => {
+            motors.push(motorIndex);
+        });
+
+        const promises = motors.map(portID => {
+            const motor = this._peripheral.motor(portID);
+            if (motor) {
+                return new Promise(resolve => {
+                    motor.turnOnForDegrees(degrees, sign);
+                    motor.pendingPromiseFunction = resolve;
+                });
+            }
+            return null;
+        });
+        // To prevent the block from returning a value, an empty function is added to the .then
+        return Promise.all(promises).then(() => {});
+    }
+
+    /**
+     * Turn specified motor(s) on indefinitely.
+     * @param {object} args - the block's arguments.
+     * @property {MotorID} MOTOR_ID - the motor(s) to activate.
+     * @return {Promise} - a Promise that resolves after some delay.
+     */
+    motorOn (args) {
+        // TODO: cast args.MOTOR_ID?
+        this._forEachMotor(args.MOTOR_ID, motorIndex => {
+            const motor = this._peripheral.motor(motorIndex);
+            if (motor) {
+                motor.turnOn();
+            }
+        });
+
+        return new Promise(resolve => {
+            window.setTimeout(() => {
+                resolve();
+            }, BoostBLE.sendInterval);
+        });
+    }
+
+    /**
+     * Turn specified motor(s) off.
+     * @param {object} args - the block's arguments.
+     * @property {MotorID} MOTOR_ID - the motor(s) to deactivate.
+     * @return {Promise} - a Promise that resolves after some delay.
+     */
+    motorOff (args) {
+        // TODO: cast args.MOTOR_ID?
+        this._forEachMotor(args.MOTOR_ID, motorIndex => {
+            const motor = this._peripheral.motor(motorIndex);
+            if (motor) {
+                motor.turnOff();
+            }
+        });
+
+        return new Promise(resolve => {
+            window.setTimeout(() => {
+                resolve();
+            }, BoostBLE.sendInterval);
+        });
+    }
+
+    /**
+     * Set the power level of the specified motor(s).
+     * @param {object} args - the block's arguments.
+     * @property {MotorID} MOTOR_ID - the motor(s) to be affected.
+     * @property {int} POWER - the new power level for the motor(s).
+     */
+    setMotorPower (args) {
+        // TODO: cast args.MOTOR_ID?
+        this._forEachMotor(args.MOTOR_ID, motorIndex => {
+            const motor = this._peripheral.motor(motorIndex);
+            if (motor) {
+                motor.power = MathUtil.clamp(Cast.toNumber(args.POWER), 0, 100);
+                if (motor.isOn) {
+                    if (motor.pendingTimeoutDelay) {
+                        motor.turnOnFor(motor.pendingTimeoutStartTime + motor.pendingTimeoutDelay - Date.now());
+                    } else if (motor.pendingPositionDestination) {
```

Comment:
Is the glitch I found in going from `10%` to `100%` maybe found in here somewhere?  I'm not studying the math carefully, so just guessing.